timelock-corruption-detection/src/main/java/com/palantir/timelock/history/PaxosLogHistoryProvider.java

```diff
@@ -205,7 +219,16 @@ private PaxosHistoryOnRemote fetchHistoryFromRemote(
     }
 
     private Map.Entry<NamespaceAndUseCase, HistoryQuery> buildHistoryQuery(
-            NamespaceAndUseCase namespaceAndUseCase, Long seq) {
-        return Maps.immutableEntry(namespaceAndUseCase, HistoryQuery.of(namespaceAndUseCase, seq));
+            NamespaceAndUseCase namespaceAndUseCase, SequenceBounds bounds) {
+        return Maps.immutableEntry(
+                namespaceAndUseCase, HistoryQuery.of(namespaceAndUseCase, bounds.lower(), bounds.upper()));
+    }
+
+    private void updateProgressState(Map<NamespaceAndUseCase, SequenceBounds> namespaceAndUseCaseSequenceBoundsMap) {
+        namespaceAndUseCaseSequenceBoundsMap.forEach((key, value) -> {
+            long lastVerifiedSequence = value.upper();
+            verificationProgressStateCache.put(key, lastVerifiedSequence);
+            logVerificationProgressState.updateProgress(key.namespace(), key.useCase(), lastVerifiedSequence);
+        });
```

Comment:
Do this in a different thread along with reset progress? 

Logic for resetting  - fetch latest seq number for the namespace useCase tuple and see if it is less than lastVerified?